bot/resources/tags/sql-fstring.md

```diff
@@ -0,0 +1,16 @@
+**SQL & f-strings**
+Don't use f-strings (`f""`) or other forms of "string interpolation" (`%`, `+`, `.format`) to inject data into a SQL query. It is an endless source of bugs and syntax errors. Additionally, in user-facing applications, it presents a major security risk via SQL injection.
+
+Your database library should support "query parameters". A query parameter is a placeholder that you put in the SQL query. When the query is executed, you provide data to the database library, and the library inserts the data into the query for you, **safely**.
+
+For example, the sqlite3 package supports using `?` as a placeholder:
+```py
+query = "SELECT * FROM stocks WHERE symbol = ?;"
+params = ("RHAT",)
+db.execute(query, params)
```

Comment:
Should we be showing what the end result becomes in a comment?

Reply:
So I think that would make the tag a bit too lengthy, which is why I'm opting not to. In my ideal world I'd like to show how a the SQL statement would differ between the params and the f-strings, but at that point it's more appropriate for an article.

I'm hoping it's clear enough to get the point across with someone (in channel) explaining or to have enough of a starting point for them to investigate what's happening in more detail with better search terms.